ambry-router/src/test/java/com.github.ambry.router/CryptoJobHandlerTest.java

```diff
@@ -178,49 +178,56 @@ public void testPendingEncryptJobs() throws InterruptedException, GeneralSecurit
     List<EncryptJob> encryptJobs = new ArrayList<>();
     SecretKeySpec perBlobKey = kms.getRandomKey();
     for (int i = 0; i < testDataCount; i++) {
-      Pair<BlobId, ByteBuffer> randomData = getRandomBlob(referenceClusterMap);
+      TestBlobData testData = getRandomBlob(referenceClusterMap);
       if (i < closeOnCount) {
-        cryptoJobExecutorService.submitJob(
-            new EncryptJob(randomData.getFirst(), randomData.getSecond(), perBlobKey, cryptoService, kms,
+        cryptoJobHandler.submitJob(
+            new EncryptJob(testData.blobId.getAccountId(), testData.blobId.getContainerId(), testData.blobContent,
+                testData.userMetadata, perBlobKey, cryptoService, kms,
                 (EncryptJob.EncryptJobResult result, Exception exception) -> {
                   encryptCallBackCount.countDown();
-                  Assert.assertEquals("BlobId mismatch ", randomData.getFirst(), result.getBlobId());
-                  Assert.assertNotNull("Encrypted content should not be null", result.getEncryptedContent());
+                  Assert.assertNotNull("Encrypted content should not be null", result.getEncryptedBlobContent());
+                  Assert.assertNotNull("Encrypted user-metadata should not be null", result.getEncryptedUserMetadata());
                   Assert.assertNotNull("Encrypted key should not be null", result.getEncryptedKey());
                 }));
       } else {
-        encryptJobs.add(new EncryptJob(randomData.getFirst(), randomData.getSecond(), perBlobKey, cryptoService, kms,
-            (EncryptJob.EncryptJobResult result, Exception exception) -> {
-              encryptCallBackCount.countDown();
-              Assert.assertEquals("BlobId mismatch ", randomData.getFirst(), result.getBlobId());
-              if (exception == null) {
-                Assert.assertNotNull("Encrypted content should not be null", result.getEncryptedContent());
-                Assert.assertNotNull("Encrypted key should not be null", result.getEncryptedKey());
-              } else {
-                Assert.assertTrue("Exception cause should have been GeneralSecurityException",
-                    exception instanceof GeneralSecurityException);
-                Assert.assertNull("Encrypted contents should have been null", result.getEncryptedContent());
-                Assert.assertNull("Encrypted key should have been null", result.getEncryptedKey());
-              }
-            }));
+        encryptJobs.add(
```

Comment:
Have you seen that failing these asserts will actually fail the test?  I've seen problems in the past where junit asserts done in threads that aren't the main thread don't actually fail the test if they fail.